Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/mysql/MySQLMvpModel.java

```diff
@@ -0,0 +1,301 @@
+        /*Sku sku = new Sku().withName("GP_Gen5_4");*/
+        Server result = manager.servers().define(serverName).withRegion(region.name()).withExistingResourceGroup(resourceGroupName)
+                .withProperties(properties)/*.withSku(sku)*/.create();
+        return result;
+    }
+
+    public static void delete(final String subscriptionId, final String id) {
+        final MySQLManager mySQLManager = AuthMethodManager.getInstance().getMySQLManager(subscriptionId);
+        mySQLManager.servers().deleteByIds(id);
+    }
+
+    public static void start(final String subscriptionId, final Server server) {
+        Server currentServer = MySQLMvpModel.findServer(subscriptionId, server.resourceGroupName(), server.name());
+        Preconditions.checkArgument(
+                ServerState.fromString("Stopped").equals(currentServer.userVisibleState()) || ServerState.DISABLED.equals(currentServer.userVisibleState()),
+                "Start action is not supported for non-disabled server.");
+        final MySQLManager mySQLManager = AuthMethodManager.getInstance().getMySQLManager(subscriptionId);
+        mySQLManager.servers().inner().start(server.resourceGroupName(), server.name());
+    }
+
+    public static void restart(final String subscriptionId, final Server server) {
+        Server currentServer = MySQLMvpModel.findServer(subscriptionId, server.resourceGroupName(), server.name());
+        Preconditions.checkArgument(ServerState.READY.equals(currentServer.userVisibleState()), "Restart action is not supported for non-ready server.");
+        final MySQLManager mySQLManager = AuthMethodManager.getInstance().getMySQLManager(subscriptionId);
+        mySQLManager.servers().inner().restart(server.resourceGroupName(), server.name());
+    }
+
+    public static void stop(final String subscriptionId, final Server server) {
+        Server currentServer = MySQLMvpModel.findServer(subscriptionId, server.resourceGroupName(), server.name());
+        Preconditions.checkArgument(ServerState.READY.equals(currentServer.userVisibleState()), "Stop action is not supported for non-ready server.");
+        final MySQLManager mySQLManager = AuthMethodManager.getInstance().getMySQLManager(subscriptionId);
+        mySQLManager.servers().inner().stop(server.resourceGroupName(), server.name());
+    }
+
+    public static List<String> listSupportedVersions() {
+        List<String> resultList = new ArrayList();
+        resultList.add(ServerVersion.FIVE_FULL_STOP_SIX.toString());
+        resultList.add(ServerVersion.FIVE_FULL_STOP_SEVEN.toString());
+        resultList.add(ServerVersion.EIGHT_FULL_STOP_ZERO.toString());
+        return resultList;
+    }
+
+    public static List<Region> listSupportedRegions() {
+        return Arrays.asList(com.microsoft.azure.arm.resources.Region.values()).stream()
+                .filter(e -> MYSQL_SUPPORTED_REGIONS.contains(e.name()))
+                .map(e -> Region.findByLabelOrName(e.name()))
```

Comment:
So we need to filter regions both in `MYSQL_SUPPORTED_REGIONS` and `com.microsoft.azure.arm.resources.Region.values()`?

Reply:
yes.